var/spack/repos/builtin/packages/cntk/package.py

```diff
@@ -49,6 +54,26 @@ class Cntk(Package):
     # Patch to change behaviour of lock file - https://github.com/Microsoft/CNTK/issues/62
     patch('lock-file.patch')
 
+    # It seems that cntk, at least version 2.0, can not be built with GCC
+    # beyond 4.8.5.
+    conflicts('%gcc@5:')
+
+    def patch(self):
+        if 'protobuf+shared' in self.spec:
+            library_suffix = 'so'
+        else:
+            library_suffix = 'a'
+
+        protobuf_libdir = os.path.basename(self.spec['protobuf'].libs.directories[0])
+        protobuf_ld_flags = self.spec['protobuf'].libs.ld_flags
+
+        filter_file(r'(protobuf_check=)lib/(libprotobuf\.)a',
+                    r'\1{0}/\2{1}'.format(protobuf_libdir, library_suffix),
```

Comment:
You should be able to use `spec['protobuf'].libs[0]` if you just want the full path to the library file

Reply:
Due to the way the check is in the configure script, it needs to be split. But your comment gave me an idea of how to do this better and eliminate the need for the suffix variable.